astropy/time/time_helper/function_helpers.py

```diff
@@ -28,3 +35,36 @@ def linspace(tstart, tstop, *args, **kwargs):
     else:
         offsets = np.linspace(np.zeros(tstart.shape), np.ones(tstop.shape), *args, **kwargs)
         return tstart + (tstop - tstart) * offsets
+
+
+@custom_functions(helps={np.mean})
+def mean(
+        a: 'astropy.time.Time',
+        axis: Optional[Union[int, Tuple[int]]] = None,
+        # dtype: numpy.typing.DTypeLike = None,
+        # out: Optional[astropy.time.Time] = None,
+        keepdims: bool = np._NoValue,
+        *,
+        where: 'numpy.typing.NDArray[bool]' = np._NoValue,
+):
+    import astropy.time
+    result = np.mean(
+        a=a.to_value(format='unix'),
+        axis=axis,
+        # dtype=dtype,
+        # out=out,
+        keepdims=keepdims,
+        where=where,
+    )
```

Comment:
@mhvk is the expert in making sure `Time` maintains full precision, but I think here that you want to get the JD values `a.jd1` and `a.jd2` and take the mean of those separately. Then make the new time using
```
result = Time(jd1_mean, jd2_mean, format='jd', ...)
result.format = a.format
```
By using the `unix` format you degrade the time precision from 128 bits to 64 bits.

Reply:
@taldcroft, thanks for your comments, I had no idea about the 64 vs. 128 bit formats. I've updated the pull request with your suggestion.